pkg/controllers/deprovisioning/helpers.go

```diff
@@ -177,42 +180,43 @@ func disruptionCost(ctx context.Context, pods []*v1.Pod) float64 {
 	return cost
 }
 
-// GetCandidates returns nodes that appear to be currently deprovisionable based off of their provisioner
+// GetCandidates returns nodes that appear to be currently deprovisionable based off of their nodePool
 func GetCandidates(ctx context.Context, cluster *state.Cluster, kubeClient client.Client, recorder events.Recorder, clk clock.Clock, cloudProvider cloudprovider.CloudProvider, shouldDeprovision CandidateFilter) ([]*Candidate, error) {
-	provisionerMap, provisionerToInstanceTypes, err := buildProvisionerMap(ctx, kubeClient, cloudProvider)
+	nodePoolMap, nodePoolToInstanceTypesMap, err := buildNodePoolMap(ctx, kubeClient, cloudProvider)
 	if err != nil {
 		return nil, err
 	}
 	candidates := lo.FilterMap(cluster.Nodes(), func(n *state.StateNode, _ int) (*Candidate, bool) {
-		cn, e := NewCandidate(ctx, kubeClient, recorder, clk, n, provisionerMap, provisionerToInstanceTypes)
+		cn, e := NewCandidate(ctx, kubeClient, recorder, clk, n, nodePoolMap, nodePoolToInstanceTypesMap)
 		return cn, e == nil
 	})
 	// Filter only the valid candidates that we should deprovision
 	return lo.Filter(candidates, func(c *Candidate, _ int) bool { return shouldDeprovision(ctx, c) }), nil
 }
 
-// buildProvisionerMap builds a provName -> provisioner map and a provName -> instanceName -> instance type map
-func buildProvisionerMap(ctx context.Context, kubeClient client.Client, cloudProvider cloudprovider.CloudProvider) (map[string]*v1alpha5.Provisioner, map[string]map[string]*cloudprovider.InstanceType, error) {
-	provisioners := map[string]*v1alpha5.Provisioner{}
-	var provList v1alpha5.ProvisionerList
-	if err := kubeClient.List(ctx, &provList); err != nil {
-		return nil, nil, fmt.Errorf("listing provisioners, %w", err)
+// buildNodePoolMap builds a provName -> nodePool map and a provName -> instanceName -> instance type map
+func buildNodePoolMap(ctx context.Context, kubeClient client.Client, cloudProvider cloudprovider.CloudProvider) (map[nodepoolutil.Key]*v1beta1.NodePool, map[nodepoolutil.Key]map[string]*cloudprovider.InstanceType, error) {
```

Comment:
how does this impact our memory footprint? I realize it'll be only one version, but IIRC, building the instance type map is pretty big.

Reply:
It doesn't change our footprint. We are currently building a `ProvisionerToInstanceType` map right now, which means that this is only a name change that doesn't tangibly change the current memory footprint.